### PR TITLE
Replace `.` with `(.|\n)` to truly allow ANY character after a backslash

### PR DESCRIPTION
**Pull request by DavisVaughan**
@kevinushey my proposed alternative of `/'((?:\\')|[^'])*'/` actually wasn't quite right, because it allowed `'\'` and thought that was a string, when really that's an unclosed single quote due to the escaped `\'`.

So I think the original regex was mostly right, we just need to also account for `\n` since the handling of `.` is explicitly written as "every character except `\n`" in the tree sitter regex handler:
https://github.com/tree-sitter/tree-sitter/blob/d92a7a969099a0a2db61a0ba5e6275e71e09c637/cli/src/generate/prepare_grammar/expand_tokens.rs#L249-L251

I'm copying the bash implementation here with `(.|\n)`
https://github.com/tree-sitter/tree-sitter-bash/blob/975bc70ad95dbbf2733872bc2e0a059c055db983/grammar.js#L837

It seems like you would typically use a `/s` (DOTALL) flag, or something like `[^]`, to accomplish this, but neither of those work in the tree sitter regex implementation.

### CODE REVIEW

File: src/parser.c [hunk 1]
```diff
@@ -17409,16 +17409,13 @@ static bool ts_lex(TSLexer *lexer, TSStateId state) {
...
       END_STATE();
     case 20:
-      if (lookahead != 0 &&
-          lookahead != '\n') ADVANCE(14);
+      if (lookahead != 0) ADVANCE(14);
       END_STATE();
     case 21:
-      if (lookahead != 0 &&
-          lookahead != '\n') ADVANCE(17);
+      if (lookahead != 0) ADVANCE(17);
```
Comment (DavisVaughan) [inline src/parser.c:17418]: This parser diff gives me confidence that this was the right way to handle this

**Review by kevinushey — APPROVED** [review]
LGTM!

File: src/parser.c (hunk 1 above)
Comment (kevinushey) [reply src/parser.c:17418]: Woah -- does this mean treesitter is parsing the JS regular expressions and re-compiling them into plain old C code? That's pretty rad.

Comment (DavisVaughan) [reply src/parser.c:17418]: That's exactly what it does! It's neat until they don't support something you wanted to use 😢

**Pull request merged**